### PR TITLE
Update Cache integrations (apart from ehcache)

### PR DESCRIPTION
**Pull request by timyates**
Upgrades the following to the latest major versions

- managed-caffeine
    ~~managed-ehcache~~
- managed-hazelcast
- managed-infinispan

ehache will be done in a separate commit as it is failing here

### CODE REVIEW

File: cache-ehcache/src/test/groovy/io/micronaut/cache/ehcache/EhcacheCacheManagerFactorySpec.groovy [hunk 1]
```diff
@@ -18,8 +18,8 @@ package io.micronaut.cache.ehcache
 import io.micronaut.context.ApplicationContext
 import org.ehcache.CacheManager
 import org.ehcache.Status
+import org.ehcache.core.internal.statistics.DefaultStatisticsService
 import org.ehcache.core.spi.service.StatisticsService
-import org.ehcache.core.statistics.DefaultStatisticsService
```
Comment (timyates) [inline cache-ehcache/src/test/groovy/io/micronaut/cache/ehcache/EhcacheCacheManagerFactorySpec.groovy:22]: This class seems to have moved to be `internal` 🤔

File: cache-ehcache/src/test/groovy/io/micronaut/cache/ehcache/EhcacheCacheManagerFactorySpec.groovy [hunk 2]
```diff
@@ -43,13 +43,18 @@ class EhcacheCacheManagerFactorySpec extends Specification {
...
+        statisticsService.getCacheStatistics("foo")
+
+        when:
+        statisticsService.getCacheStatistics("not-foo")
 
         then:
-        statisticsService
-        statisticsService.started
+        def ex = thrown(IllegalArgumentException)
+        ex.message == "Unknown cache: not-foo"
```
Comment (timyates) [inline cache-ehcache/src/test/groovy/io/micronaut/cache/ehcache/EhcacheCacheManagerFactorySpec.groovy:57]: There's no `started` anymore, so I check for existence of a known cache, and that an unknown cache is missing

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
+managed-ehcache = "3.10.2"
+managed-hazelcast = "5.1.3"
+managed-infinispan = "13.0.11.Final"
 
 [libraries]
 # Managed
 managed-caffeine = { module = "com.github.ben-manes.caffeine:caffeine", version.ref = "managed-caffeine" }
 managed-hazelcast = { module = "com.hazelcast:hazelcast", version.ref = "managed-hazelcast" }
+managed-ehcache = { module = "org.ehcache:ehcache", version.ref = "managed-ehcache" }
 managed-ehcache-clustered = { module = "org.ehcache:ehcache-clustered", version.ref = "managed-ehcache" }
```
Comment (timyates) [inline gradle/libs.versions.toml:13]: I believe ehcache has changed how they package their libraries now, as we need ehcache as well as ehcache-clustered

Comment (timyates) [reply gradle/libs.versions.toml:13]: Yeah, it used to be `compile`, but now it's `runtime`

**Comment by timyates** [comment]
Investigating the failure...  Need to fire up my x86 machine (as there's no arm terracotta docker image 😿)

File: cache-ehcache/build.gradle [hunk 4]
```diff
@@ -15,6 +15,6 @@ dependencies {
 
 micronautBuild {
     resolutionStrategy {
-        force "org.ehcache:ehcache-clustered:$ehcacheVersion"
+        force "org.ehcache:ehcache-clustered:${libs.versions.managed.ehcache.get()}"
```
Comment (alvarosanchez) [inline cache-ehcache/build.gradle:18]: Perhaps we can get rid of this. I suppose it was introduced to avoid dependabot trying to upgrade it

/cc @melix

**Review by alvarosanchez — APPROVED** [review]

File: cache-ehcache/build.gradle (hunk 4 above)
Comment (timyates) [reply cache-ehcache/build.gradle:18]: I'll do that as part of the ehcache upgrade here https://github.com/micronaut-projects/micronaut-cache/issues/523

**Pull request merged**